packages/adapters-library/src/defiProvider.ts

```diff
@@ -808,6 +841,12 @@ export class DefiProvider {
         blockNumber,
       })
 
+      if (tokens.length > 1) {
+        throw new TvlValidationError(
+          'Total value locked should return maximum one protocol token, adapter incorrectly implemented',
+        )
+      }
+
```

Comment:
If protocol token length is more than one that mean the adapter has not applied the protocol token address filter